### DISCUSSION TITLE
Nanopore PCR-cDNA RNA seq, quality checks prior running Isoquant

### DISCUSSION THREAD

**Discussion by Mvirale** [ablab/IsoQuant#237] [Q&A]
Hi Team, 

Thanks for the fantastic tool!
I have RNA-seq data obtained with the Nanopore PCR-cDNA Barcoding Kit. 
Before running Isoquant, do you recommend correcting the reads using Medaka/Racon/Falcon? 
Do you suggest orienting reads and removing the adapters and barcodes using Pychopper or similar tools?
Can the use of Pychopper affect the isoform discovery process?

Thanks in advance for the help.
Best

**Comment by andrewprzh** [comment]
Dear @Mvirale

Thank you for the feedback!

I would say IsoQuant doesn't need any error correction to be done prior to running it. IsoQuant uses minimap2, which in my opinion handles high error rate quite well. Furthermore, IsoQuant is also designed to deal with Nanopore sequencing errors and implements a separate algorithm for correcting inaccurate spliced alignments. Orienting reads is also not required.

With respect to trimming, I never tried Pychopper (although it's still on my list) and I always processed barcoded cDNA ONT data as is. I feel like minimap2 simply clips non-genomic part (polyA, barcodes, adapters) and the resulting alignments are reliable enough to feed them into IsoQuant. Also, IsoQuant benefits from the presence of polyA tails since they help to detect transcript ends and APA sites.

If you (or anyone) have any information about how trimming affects the alignment quality and IsoQuant results - I'd be happy to hear it!

Best
Andrey

**Reply by Mvirale** [reply]
Dear Andrey, 

Thanks a lot for all the info!

Best
Matteo
